archey/entries/terminal.py

```diff
@@ -1,11 +1,58 @@
 """Terminal detection class"""
 
 import os
+import re
 
 from archey.colors import Colors
 from archey.entry import Entry
 
 
+# We detect a terminal by using the following three constants in the order below:
+# First, we try using the value in the `TERM_PROGRAM` enviroment variable.
+# Then, we use `COLORTERM_DICT` to try matching a value with the `COLORTERM` environment variable.
+# Third, we use `TERM_DICT` to try matching a value with the $TERM environment variable.
+# Finally, we fall back to custom environment variables defined in `ENV_DICT`.
+# If none of the above tests find a value, we use whichever value was defined in `$TERM`.
+
+# All of the keys in `COLORTERM_DICT` and `TERM_DICT` are matched as regular expressions...
+#   (using `re.match`, i.e. attempting to match once at the beginning of the string), so be careful!
```

Comment:
This is just to help any users who want to add support for their favourite terminal emulator 😄